src/data-nodes/shared-memory/compute.jl

```diff
@@ -0,0 +1,97 @@
+"""
+Compute node for Files/IO.
+Computing these will result in a `SharedMemory` data node with
+the offsets on the data node as specified by the Partition.
+"""
+
+immutable FileBlocks{P <: AbstractPartition} <: ComputeNode
+    uri::AbstractString #Change to URI
+    partition::P
+end
+
+FileBlocks(uri::AbstractString) = FileBlocks(uri, BytesPartition{nworkers()}())
+
+"""
+Data Node for Shared Memory. (Dist File?)
+This contains the mmaped file handle, the offset mapping to the processes and the
+partition.
+This node basically means the data has been created into an mmaped file and we
+have a pid -> offset/BlockIO ref mapping set up.
+"""
+immutable SharedMemory{P<:AbstractPartition} <: DataNode
+    sharedarray::Array #Entire mmaped file
+    offsets::Vector #Mapping btwn PIDs and the offsets they are responsible for.
+    refs::Vector #RemoteRefs to the mmaps on each process.
+    partition::P
+end
+
+function gather(ctx, n::SharedMemory)
+    #This returns the entire mmap. Partition does not matter.
+    #Write to a file and return the file handle here?
+    n.sharedarray
+end
+
+##### Compute #####
+
+"""
+Computing a FileBlock, will create a SharedMemory with the specified partition.
+"""
+function compute(ctx, x::FileBlocks)
```

Comment:
Should `compute` always return `SharedMemory` datanodes?
How does this work when the workers are not on same machine?